tools/ALARAJOYWrapper/preprocess_fendl3.py

```diff
@@ -2,81 +2,178 @@
 import reaction_data as rxd
 import tendl_processing as tp
 import njoy_tools as njt
+import numpy as np
 import argparse
 import warnings
 from pathlib import Path
+from collections import defaultdict
 
 def args():
     parser = argparse.ArgumentParser()
     parser.add_argument(
         '--fendlFileDir', '-f', required=False, nargs=1
     )
+    parser.add_argument(
+        '--gas_handling', '-g', required=True, nargs=1
+    )
     parser.add_argument(
         # Temperature for NJOY run [Kelvin]
         '--temperature', '-t', required=False, nargs=1, default=[293.16]
     )
     return parser.parse_args()
 
-def write_dsv(dsv_path, cumulative_data):
+def remove_gas_daughters(all_rxns):
+    """
+    Remove reactions from the dictionary that produce a light gas daughter
+        (i.e. a nuclide lighter than an alpha particle) whose total gas
+        production is otherwise accounted for by the MT = 203-207 "reactions"
+        to avoid double-counting.
+
+        Optional method to be called within gas_handling().
+
+    Arguments:
+        all_rxns (collections.defaultdict): Hierarchical dictionary keyed by
+            parent nuclides to store all reaction data, with structured as:
+            {parent:
+                {daughter:
+                    {MT:
+                        {
+                            'emitted': (str of emitted particles)
+                            'non_zero_groups': (int of non-zero groupwise XS)
+                            'xsections': (array of groupwise XS)
+                        }
+                    }
+                }    
+            }
+    
+    Returns:
+        all_rxns (collections.defaultdict): Modified version of all_rxns with
+            double-counted gas-producing reactions left out.
+    """
+
+    for parent in all_rxns:
+        for daughter in all_rxns[parent]:
+            MTs = set(all_rxns[parent][daughter])
+            for MT in MTs:
+                gas_MTs = tp.GAS_DF['total_mt'].tolist()
+                if MT not in gas_MTs and any(MT in gas_MTs for MT in MTs):
+                    del all_rxns[parent][daughter][MT]
```

Comment:
I think I'd loop this differently  - something like this.
```suggestion
    gas_data = tp.GAS_DF['total_mt'].tolist()
    for parent, rxns in all_rxns.items():
        daughterKZAs = list(rxns.keys())
        for gasKZA, gas_MT in tp.GAS_DF['kza','total_mt']:
            if gasKZA in daughterKZAs and gas_MT in rxns[gasKZA].keys():
               # delete all non gas_MTs
```